lib/SPIRVProducerPass.cpp

```diff
@@ -1255,6 +1255,39 @@ void SPIRVProducerPass::FindTypePerFunc(Function &F) {
         continue;
       }
 
+      // #497: InsertValue and ExtractValue map to OpCompositeExtract which
+      // takes literal values for indices. As a result don't map the type of
+      // indices.
```

Comment:
you don't have to map the type of the indices because the indices are always small, by construction (various universal limits)

Reply:
The indices are not ids in OpCompositeInsert and OpCompositeExtract, they are literals.